Content.Shared/Chat/TypingIndicator/TypingIndicatorEvents.cs

```diff
@@ -16,3 +17,17 @@ public TypingChangedEvent(bool isTyping)
         IsTyping = isTyping;
     }
 }
+
+[Serializable, NetSerializable]
+public sealed class BeforeShowTypingIndicatorEvent : IInventoryRelayEvent
+{
+    public SlotFlags TargetSlots { get; } = SlotFlags.WITHOUT_POCKET;
+    public string? OverrideIndicator = null;
+    public TimeSpan? LatestEquipTime = null;
```

Comment:
I would make these fields private and add a public method that handles the logic of comparing equip times and setting these fields.